pep-0497-externals.txt

```diff
@@ -0,0 +1,252 @@
+PEP: 497
+Title: Enhanced handling of external requirements
+Version: $Revision$
+Last-Modified: $Date$
+Author: Tennessee Leeuwenburg <tleeuwenburg@gmail.com>,
+
+Status: Draft
+Type: Standards Track
+Python-Version 3.6
+Content-Type: text/x-rst
+Created: 03-Aug-2015
+Post-History: 03-Aug-2015
+
+
+Abstract
+========
+
+The canonical method for installing packages is to use pip, whether that is
+referencing PyPI or a local repository. A limitation of the current install
+ecosystem is the relationship with operating-system level dependencies. This
+can result in compilation failures or execution failures. Package maintainers
+may wish to specify the requirements to avoid this issue, and users may
+appreciate earlier stopping and more informative messages about their system.
+
+The responsibility of pip is to stop early and provide useful information about
+missing and not automatically installable dependecies. Ideally it would provide
+a useful user message about how to obtain the required dependencies. The
+responsibility for actual installation lies with the user at this time.
+
+This PEP specifies an extension to the requirements sections of setup.py
+which will allow authors to express these requirements cleanly. In the
+implementation section, a discussion of how this is used to solved problems
+is given.
+
+Here is an example of the proposed syntax, where both operating system and build
+target specifiers are included.
+
+::
+
+    setup(
+        name='uses_scipy_with_atlas_extension'
+        install_requires='yaml, scipy[atlas]',
+        setup_requires='pyfortran',
+        external={
+          setup=[
+            'bin!fortran:dist_name=='ubuntu',
+            'include!kernel.h',
+            'include!liblas.h',
+          ],
+          install=[
+              'lib!liblas1.so'
+          ],
```

Comment:
I'm not opposed, but I don't think this is obvious either. The original version puts the external deps into a clearly separated area for the purpose; your suggestion pulls the two together. This may be conflating two things
event though it makes it shorter. Perhaps it is worth having a quick chat to decide which version goes in the abstract, and then including the alternative later in the document for discussion.